frontend/src/components/Module.jsx

```diff
@@ -0,0 +1,125 @@
+import { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import {
+  Checkbox,
+  ClickAwayListener,
+  IconButton,
+  TextField,
+} from "@mui/material";
+import EditIcon from "@mui/icons-material/Edit";
+import DeleteIcon from "@mui/icons-material/Delete";
+import ConfirmDialog from "./ConfirmDialog";
+import ApiHelper from "../helpers/apiHelper";
+import loadData from "../helpers/loadData";
+
+export default function Module({ moduleId, setLoadingModal }) {
+  const [loadingModule, setLoadingModule] = useState(true);
+  const [isEditActive, setIsEditActive] = useState(false);
+  const [newName, setNewName] = useState("");
+  const [module, setModule] = useState({
+    id: 0,
+    title: "Please Wait",
+    isDone: false,
+  });
+  const [confirmOpen, setConfirmOpen] = useState(false);
+
+  useEffect(() => loadData("modules", setModule, moduleId), [loadingModule]);
+
+  const handleCheckbox = (checked) => {
+    ApiHelper(
+      `modules/${moduleId}`,
+      "patch",
+      {
+        isDone: checked,
+      },
+      "application/merge-patch+json"
+    )
+      .then(() => {
+        setLoadingModule(!loadingModule);
+      })
+      .catch((err) => {
+        console.error(`Axios Error : ${err.message}`);
+      });
+  };
+
+  const editModule = (newModuleName) => {
```

Comment:
Add handle to your const name, like "handleEditModule", better.

Reply:
Change the name in handleEditModule